drivers/flash/flash_stm32_ospi.c

```diff
@@ -952,6 +946,13 @@ static int flash_stm32_ospi_erase(const struct device *dev, off_t addr,
 					: HAL_OSPI_INSTRUCTION_8_BITS;
 
 	while ((size > 0) && (ret == 0)) {
+
+		if (stm32_ospi_write_enable(&dev_data->hospi,
+			dev_cfg->data_mode, dev_cfg->data_rate) != 0) {
+			LOG_ERR("Erase failed : write enable");
+			return -EIO;
```

Comment:
Could you check what is the ospi_lock_thread/ospi_unlock_thread  position when  we are returning with error in the middle of the while loop. 
Before this change, I  agree that ospi_lock_thread(dev); should have been  placed _after_ the stm32_ospi_mem_ready
but inside the while loop, using break and ret code will avoid returning without  ospi_unlock_thread()

Note that during flash_stm32_ospi_write, ospi_lock_thread should also be after  stm32_ospi_mem_ready()

Reply:
> Feels like you are doing one-too many things as part of a single commit, 

Especially changing the sample is a good idea of enhancement, because it helps demonstrating the bug/correction.
I agree that this is at least another commit, as it affects other than stm32 driver. 
This change (more sectors) could be proposed as a new items or options in the sample.

Reply:
So, I should revert the changes in the sample and introduce a new function in the sample that test multi sector erase protect this as compile option. Then split the pr into the bug itself and the change in the sample. 
Is this correctly understood?